core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -144,16 +144,18 @@ public Schema array(Schema array, Schema element) {
       if (selectedIds.contains(keyId) || selectedIds.contains(valueId)) {
         return complexMapWithIds(array, keyId, valueId);
       } else if (element != null) {
-        Schema keyProjection = element.getField("key").schema();
+        Schema.Field keyProjectionField = element.getField("key");
```

Comment:
Hi @lxynov,  could you write a test to reproduce the issue? I don't completely understand it.  The description says that this fixes pruning if a  map's value is a complex type [nested], but the change is mostly a null check on the key field